src/App.tsx

```diff
@@ -1,14 +1,38 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { SelectCategory } from './types/SelectCategory';
+import { Todo } from './types/Todo';
+import { filterTodos } from './components/utils/filterTodos';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [query, setQuery] = useState('');
+  const [selectTodoId, setSelectTodoId] = useState<number | null>(null);
+  const [selectCategory, setSelectCategory] = useState<SelectCategory>(
+    SelectCategory.all,
+  );
+
+  const selectedTodo = todos.find(todo => todo.id === selectTodoId);
+
+  const filteredTodos = filterTodos(todos, { selectCategory, query });
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getTodos()
+      .then(data => setTodos(data))
+      .finally(() => setIsLoading(false));
```

Comment:
The `getTodos` function call should include a `catch` block to handle potential errors. This is important to avoid unhandled promise rejections and to provide feedback to the user in case of an error.